model/ftn/wmesmfmd.ftn

```diff
@@ -6551,6 +6986,191 @@
+!
+!      Name      Type  Module   Description
+!     ----------------------------------------------------------------
+!      NONE
+!     ----------------------------------------------------------------
+!
+!  5. Called by :
+!
+!  6. Error messages :
+!
+!  7. Remarks :
+!
+!  8. Structure :
+!
+!  9. Switches :
+!
+! 10. Source code :
+!
+!/ ------------------------------------------------------------------- /
+!/
+      USE W3FLDSMD, ONLY: W3FLDO, W3FLDG
+      USE WMUNITMD, ONLY: WMUGET, WMUSET
+      IMPLICIT NONE
+!/ ------------------------------------------------------------------- /
+!/ Parameter list
+!/
+      character(len=3), intent(inout)  :: idfld
+      type(ESMF_Field), intent(inout)  :: fldwx
+      type(ESMF_Field), intent(inout)  :: fldwy
+      integer, intent(in)              :: time0(2)
+      integer, intent(in)              :: timen(2)
+      integer, intent(inout), optional :: rc
+!/
+!/ ------------------------------------------------------------------- /
+!/ Local parameters
+!/
+      integer :: ierr, tw0l(2), twnl(2), lb(2), ub(2)
+      real :: wx0l(nx,ny), wy0l(nx,ny)
+      real :: wxnl(nx,ny), wynl(nx,ny)
+      real :: dt0l(nx,ny), dtnl(nx,ny)
+      real(ESMF_KIND_RX), pointer :: dptr(:,:)
+      integer :: mdse = 6
+      integer :: mdst = 10
+      integer, save :: mdsf
+      character(256) :: logmsg
+      logical :: flagsc = .false.
+      integer, parameter :: lde = 0
+      logical, save :: firstCall = .true.
+      character(len=13) :: tsstr
+      character(len=3) :: tsfld
+      integer :: nxt, nyt, gtypet, filler(3), tideflag
+#if defined(TEST_WMESMFMD) || defined(TEST_WMESMFMD_READFROMFILE)
+      integer, save :: timeSlice = 1
+#endif
+!
+! -------------------------------------------------------------------- /
+!
+      rc = ESMF_SUCCESS
+
+      if (firstCall) then 
+        ! assign unit number for input file 
+        call wmuget(mdse, mdst, mdsf, 'INP')
+        call wmuset(mdse, mdst, mdsf, .true., desc='Input data file')
+
+        ! open file      
+        call w3fldo('READ', idfld, mdsf, mdst, mdse, nx, ny, gtype, ierr)
+        if (ierr.ne.0) then
+          write(logmsg,*) "Error in opening "//idfld//", iostat = ", ierr
+          call ESMF_LogWrite(trim(logmsg), ESMF_LOGMSG_ERROR)
+          rc = ESMF_FAILURE
+          return
+        endif
+
+        firstCall = .false.
+      end if
+
+      ! init variables
+      wx0l = 0.0
+      wy0l = 0.0
+      dt0l = 0.0
+      wxnl = 0.0
+      wynl = 0.0
+      dtnl = 0.0
+
+      ! need to rewind to the begining of the file to access
+      ! data of requested date correctly 
+      rewind(mdsf)
+
+      ! read header information
+      ! this was inside of w3fldo call but since we are opening file
+      ! once and rewinding, the header need to be read
+      read(mdsf, iostat=ierr) tsstr, tsfld, nxt, nyt, &
+        gtypet, filler(1:2), tideflag
+
+      ! read input
+      call w3fldg('READ', idfld, mdsf, mdst, mdse, nx, ny, &
```

Comment:
look into if there is time interpolation

Reply:
@JessicaMeixner-NOAA if you look at following line of code https://github.com/NOAA-EMC/WW3/blob/3740ba99caaa2822b5098bf32faeaf199a710a51/model/ftn/w3fldsmd.ftn#L1118 the time interpolation is not enabled intentionally with the coupling (`FLINTERP = .FALSE.`). I did not test it but it seems like that. So, if we want to force it and if it is really related with the time interpolation then we need to set `FLINTERP = .TRUE.`. Let me know what do you think?

Reply:
I am testing to set `flagsc = .true.` in w3fldg call. This will activate the time interpolation. I'll update you with the results.

Reply:
I think that setting `flagsc = .false.` is fine and this will activate the time interpolation if the field is 'CUR', 'WND' (our case) 'WNS', 'TAU' or 'RHO'.  So, this is current behavior in the model but if you look at the following figure (red is from the point in the active coupling region and white is from data - outside of the active coupling region),

![Screen Shot 2021-06-24 at 3 31 05 PM](https://user-images.githubusercontent.com/1228999/123335148-4983f380-d501-11eb-86d4-caa92ba42655.png)

and it seem that there is no time interpolation (white one) and it is keeping constant in 3-hourly interval which I think it is the temporal resolution of the wind forcing. Anyway, I am plaining to put some print statement to see the value of `FLINTERP`. Another possibility could be having same data in current and next time and then the temohtal interpolation will produce constant value for 3-hours. I'll also check that one too.

Reply:
I print out `FLCOUPL` and `FLINTERP`. The `FLCOUPL` is set to False as expected and consistent with the `flagsc = .false.`. The `FLINTERP` seems to set as True. So, I was expecting to have time interpolation for wind at this point but previous plot does not support it. Maybe I am missing something because it is hard to follow `W3FLDG` call. @JessicaMeixner-NOAA do you have any idea about it.

Reply:
I also checked the w*0 and w*n fields read from the forcing file and their min and max values. It seems that they are fine. So, I think that read call gets the current and next time steps correctly from file and maybe there is a issue in time interpolation. 

```
20210624 172141.669 INFO             PET1340    time0 =     20190829           0
20210624 172141.669 INFO             PET1340    timen =     20190829         600
20210624 172141.669 INFO             PET1340    tw0 =     20190829           0
20210624 172141.669 INFO             PET1340    twn =     20190829       30000
20210624 172141.670 INFO             PET1340    wx0 min, max =   -19.55906       14.35927
20210624 172141.670 INFO             PET1340    wy0 min, max =   -16.34938       19.25990
20210624 172141.671 INFO             PET1340    wxn min, max =   -22.29049       13.06400
20210624 172141.671 INFO             PET1340    wyn min, max =   -15.75515       20.08858
```

Also time interval considered (time0 and timen) and field times (tw0 and twn) are fine too. I am not sure about the time interpolation logic and it seems that we need to look at it more detailed way.

Reply:
Thanks for this explanation and the rundir.  I'll look at this today to make more sense of it.  Usually for WW3 when we read in from the file, we only read in a new file time for twN once we have gotten to that, and then we set tw0=twN and same for the wind values and then just get the new winds for the twn time.  The fact that this is called every time regardless and that it's set to zero each time (and not just for the first call) is what made me look twice.   Again, I'll look at the output more closely today so I can try to understand your explanations better.  Thanks again for sharing the output!

Reply:
Apologies, I did not look at this last week.    I've looked at it now and with your explanations and looking at the run directory and plotting I see that it's working.  What through me off is that typically when we are reading from a file you do not set all of the time and variables to zero each time step and we use prior information and tw01=twN when current time gets to twN and then you don't re-read those fields.  I think this might be something we could revisit if efficiency updates are need or if we decide we want to try to interpolate the wind between the snapshots in this region.

Reply:
@JessicaMeixner-NOAA No worries. I know you are too busy with the PRs etc. Yes, we could still revisit this part of code. I could not find detailed information about the time interpolation in the read routine but if somebody has more information about that we could also enable temporal interpolation. Another option could be creating wind forcing file by increasing its temporal frequency by performing temporal interpolation outside. So, I know that the default data is 3-hourly but if we could modify the script that creates the wind input and add an option to it to perform temporal interpolation to create i.e. 1 -hourly forcing file then we don't need to change anything internally in the cap. We could just use already interpolated forcing.

Reply:
Sounds good! Always nice to have a few options so we can figure out what is most efficient or most flexible, etc.